addons/l10n_it_edi/models/account_edi_format.py

```diff
@@ -368,6 +374,35 @@ def _update_invoice_from_binary(self, filename, content, extension, invoice):
                 return self._import_fattura_pa(decoded_content, invoice)
         return super()._update_invoice_from_binary(filename, content, extension, invoice)
 
+    def _l10n_it_edi_search_tax(self, company, percentage, **extra_domain):
+        default_rates = [100.00, 100.00]
+        expected_rates = extra_domain.pop('expected_rates', default_rates) or default_rates
+        conditions = [
+            ('company_id', '=', company.id),
+            ('amount', '=', percentage),
+            ('amount_type', '=', 'percent'),
+            ('type_tax_use', '=', 'purchase'),
+        ] + [(k, '=', v) for k, v in extra_domain.items() if v]
+        taxes = self.env['account.tax'].search(conditions)
+        taxes = taxes.filtered(lambda x: x.invoice_repartition_line_ids.mapped("factor_percent") == expected_rates)
+        return taxes[0] if taxes else taxes
+
+    def _l10n_it_edi_missing_tax_message(self, percentage, name=False, **extra_domain):
```

Comment:
Then do different methods for withholding and vat (not sure even if we need a special method for this)